wallet/storage/src/internal/store_tx.rs

```diff
@@ -52,13 +55,50 @@ mod well_known {
     }
 
     declare_entry!(StoreVersion: u32);
+    declare_entry!(EncryptionKeyKdfChallenge: KdfChallenge);
+}
+
+#[derive(PartialEq, Clone)]
+pub enum EncryptionState {
+    Locked,
+    Unlocked(Option<SymmetricKey>),
 }
 
 /// Read-only chainstate storage transaction
-pub struct StoreTxRo<'st, B: storage::Backend>(pub(super) storage::TransactionRo<'st, B, Schema>);
+pub struct StoreTxRo<'st, B: storage::Backend> {
+    pub(super) storage: storage::TransactionRo<'st, B, Schema>,
```

Comment:
I don't think `pub(super)` is necessary here. Besides that, generally speaking, we consider `pub(X)` to be a bad practice that can be avoided with good structuring of modules. Sometimes it simplfies things.

One more thing: If there's something you're doing temporarily in the PR, please use either `TODO(PR)` or `FIXME`. Both will trigger a CI error through a python script (can be seen in `do_checks.sh`) to remind you of them.

Reply:
that was like that from before, was not sure what was the use case,
Made the field private as it is not used from anywhere outside, and I don't think the internal storage should be touched on a transaction anyway.